thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -0,0 +1,229 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.apache.pinot.thirdeye.anomaly.task.TaskConstants;
+import org.apache.pinot.thirdeye.anomaly.task.TaskInfoFactory;
+import org.apache.pinot.thirdeye.datalayer.bao.DatasetConfigManager;
+import org.apache.pinot.thirdeye.datalayer.bao.DetectionConfigManager;
+import org.apache.pinot.thirdeye.datalayer.bao.MetricConfigManager;
+import org.apache.pinot.thirdeye.datalayer.bao.TaskManager;
+import org.apache.pinot.thirdeye.datalayer.dto.DatasetConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.DetectionConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.MetricConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.TaskDTO;
+import org.apache.pinot.thirdeye.datasource.DAORegistry;
+import org.apache.pinot.thirdeye.datasource.MetricExpression;
+import org.apache.pinot.thirdeye.datasource.MetricFunction;
+import org.apache.pinot.thirdeye.detection.DetectionPipelineTaskInfo;
+import org.apache.pinot.thirdeye.formatter.DetectionConfigFormatter;
+import org.apache.pinot.thirdeye.rootcause.impl.MetricEntity;
+import org.apache.pinot.thirdeye.util.ThirdEyeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataAvailabilityTaskScheduler implements Runnable {
+  public static final String LIX_FLAG_VALUE = "DATA_AVAILABILITY_SCHEDULE";
+  private static final Logger LOG = LoggerFactory.getLogger(DataAvailabilityTaskScheduler.class);
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private ScheduledThreadPoolExecutor executorService;
+  private long delayInSec;
+  private int notRunThresholdInMin;
+  private Map<Long, Long> lastTaskCreateTimeMap;
+
+  public DataAvailabilityTaskScheduler(long delayInSec, int notRunThresholdInMin ) {
+    this.delayInSec = delayInSec;
+    this.notRunThresholdInMin = notRunThresholdInMin;
+    this.lastTaskCreateTimeMap = new HashMap<>();
+    this.executorService = new ScheduledThreadPoolExecutor(1, r -> {
+      Thread t = Executors.defaultThreadFactory().newThread(r);
+      t.setDaemon(true);
+      return t;
+    });
+  }
+
+  @Override
+  public void run() {
+    Multimap<Set<String>, DetectionConfigDTO> dataset2DetectionMap = ArrayListMultimap.create();
+    Map<String, Long> datasetRefreshTimeMap = new HashMap<>();
+    populateRequiredMetadata(dataset2DetectionMap, datasetRefreshTimeMap);
+    Map<Long, TaskDTO> runningDetection = retrieveRunningDetection();
+    for (Set<String> datasets : dataset2DetectionMap.keySet()) {
+      for (DetectionConfigDTO detectionConfig : dataset2DetectionMap.get(datasets)) {
+        long detectionConfigId = detectionConfig.getId();
+        long lastTimestamp = detectionConfig.getLastTimestamp();
+        boolean allUpdated = datasets.stream().map(d -> datasetRefreshTimeMap.get(d) >= lastTimestamp)
```

Comment:
Need to consider multi-day granularity here. For example, detecting on the weekly aggregated value  on daily data.

Reply:
Added TODO, since we don't tackle this in this iteration.